libsplinter/src/network/peer_manager/mod.rs

```diff
@@ -109,6 +113,12 @@ impl Drop for PeerRef {
     }
 }
 
+/// An entry of temporary peers, that may connected externally, but not yet requested locally.
+pub struct TemporaryPeer {
```

Comment:
I think this is going to cause an issue with PeerInterconnect since this information is not returned with the Connection id request. So if the PeerInterconnect gets a message from a temporary peer, for example a circuit creation request, PeerInterconnect will not know who the request is from. 

The temporary peers information should be returned in ConnectionIds request as if it was a promoted peer.

Reply:
The temporary status is different from the old idea of temporary peers. In the peer manager, temporary peers are only visible internally (the one function that list them is more for tools to have some visibility into the process - like the status endpoint, for example). 

The temporary status is also only for incoming connections. Admin service will still request a peer ref to make a circuit, which is the point at which the temp peer will be promoted.

Reply:
It probably should be pub, though... I might have left that in.

Reply:
It probably should be pub, though... I might have left that in.

Reply:
After discussion, there will be a follow up PeerInterconnect PR.

Reply:
Removed the `pub` from the struct